File: src/main/java/de/thatsich/intellie/common/registries/IRegistry.java
package de.thatsich.intellie.common.registries;

/**
 All registies need to be able to register their stuff in the end

 @author thatsIch
 @date 25.03.2014. */
public interface IRegistry<T>
{
	void add ( T elem );

	/**
	 Registers that particular part
	 */
	void register ();
}

File: src/main/scala/de/thatsich/minecraft/core/registries/IRegistry.scala
package de.thatsich.minecraft.core.registries


/**
 * simple registry
 *
 * @author thatsIch
 * @since 06.04.2014.
 */
trait IRegistry[ T ]
{
	/**
	Adds an element to the registry.
	It will get registered later on

	  * @param elem New element in the registry
	  */
	def add(elem: T)

	/**
	Registers that particular part
	  */
	def register()
}
